### DISCUSSION TITLE
Invalid upsert invocation (P2014) in update scenario

### DISCUSSION THREAD

**Discussion by remjx** [prisma/prisma#5359] [Q&A] [answered]
The `create: {}` portion of my upsert works as expected, but the `update: {}` scenario generates a P2014 error.

Upsert code:

```ts
const data = {
    User: {
      connect: {
        id: user.id,
      },
    },
    expiresAt: new Date(Date.now() + ms("10m")),
    token: randomize("0", 6),
  }
  await db.passwordResetAuthorization.upsert({
    where: {
      userId: user.id,
    },
    create: data,
    update: data,
  })
```

Prisma error: 
```
Invalid `prisma.passwordResetAuthorization.upsert()` invocation:


  The change you are trying to make would violate the required relation 'PasswordResetAuthorizationToUser' between the `PasswordResetAuthorization` and `User` models.
details:
{
  code: 'P2014',
  clientVersion: '2.14.0',
  meta: {
    relation_name: 'PasswordResetAuthorizationToUser',
    model_a_name: 'PasswordResetAuthorization',
    model_b_name: 'User'
  }
}
error stack:
• index.js:78130 request
    node_modules\@prisma\client\runtime\index.js:78130:15
```

Schema:
```prisma
model PasswordResetAuthorization {
  userId        Int         @id
  User          User        @relation(fields: [userId], references: [id])
  token         String
  createdAt          DateTime  @default(now())
  expiresAt     DateTime    
  tokenAttemptsRemaining     Int     @default(3)
}
```

Any idea what could be going on here? Thank you

## Environment
Prisma version: 2.14.0
Database: PostgreSQL 12.5

**Comment by remjx** [comment, marked as answer]
I fixed it by removing the `User` from the `update` object. I suppose if a record already exists, it is already "connected", so trying to reconnect it again throws an error.

Working code: 

```ts
  const data = {
    expiresAt: new Date(Date.now() + ms("10m")),
    token: randomize("0", 6),
  }
  await db.passwordResetAuthorization.upsert({
    where: {
      userId: user.id,
    },
    create: {
      User: {
        connect: {
          id: user.id,
        },
      },
      ...data,
    },
    update: data,
  })
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
